spinoso-string/src/enc/utf8/mod.rs

```diff
@@ -287,6 +287,17 @@ impl Utf8String {
     pub fn get_char_slice(&self, range: Range<usize>) -> Option<&'_ [u8]> {
         // TODO: use fast path rejection from `get_char` here too
         let Range { start, end } = range;
+
+        if start > self.len() || start > self.char_len() {
+            return None;
+        }
+
+        match end as isize - start as isize {
+            d if d <= 0 => return Some(&[]),
```

Comment:
because we wanted to avoid a UTF-8 decode above, we don't need to check for the end before start case and don't need to cast to `isize`.